sdk/python/kfp/dsl/types.py

```diff
@@ -17,9 +17,14 @@
 
 
 class BaseType:
-	'''MetaType is a base type for all scalar and artifact types.
+	'''BaseType is a base type for all scalar and artifact types.
 	'''
-	pass
+
+	def to_dict(self) -> dict:
+		'''to_dict serializes the type instance into a python dictionary'''
+		#return {type(self).__name__: self.__dict__}
+		return {type(self).__name__: self.__dict__} if self.__dict__ else type(self).__name__
```

Comment:
Maybe change `-> dict` to `-> Union[Dict, Text]` to comply with the actual return type?

btw, when is the case where we do not have `__dict__` attribute?

Reply:
Done.

>when is the case where we do not have __dict__ attribute?

The `__dict__` attribute still exists (usually), but it can be empty when object has no user-defined attributes. 
For example the objects of the [Integer](https://github.com/kubeflow/pipelines/blob/e46a43e72ea7096dbfe1229c00ef6ae458635618/sdk/python/kfp/dsl/types.py#L30) class will have empty `__dict__`.